atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/StreamTestWithHashStreamStore.java

```diff
@@ -179,6 +185,84 @@ protected void loadSingleBlockToOutputStream(Transaction t, Long streamId, long
+            compressedStream = new LZ4CompressingInputStream(new DigestInputStream(stream, digest));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        StreamMetadata metadata = storeBlocksAndGetFinalMetadata(null, id, compressedStream, false);
+
+        ByteString hashByteString = ByteString.copyFrom(digest.digest());
+        StreamMetadata correctedMetadata = StreamMetadata.newBuilder(metadata)
+                .setHash(hashByteString)
+                .build();
+        // Store the corrected metadata with the correct hash
+        storeMetadataAndIndex(id, correctedMetadata);
+
+        return Pair.create(id, new Sha256Hash(correctedMetadata.getHash().toByteArray()));
+    }
+
+    @Override
+    public Map<Long, Sha256Hash> storeStreams(final Transaction t, final Map<Long, InputStream> streams) {
+        if (streams.isEmpty()) {
+            return ImmutableMap.of();
+        }
+
+        Map<Long, StreamMetadata> idsToEmptyMetadata = Maps.transformValues(streams, Functions.constant(getEmptyMetadata()));
+        putMetadataAndHashIndexTask(t, idsToEmptyMetadata);
+
+        Map<Long, StreamMetadata> idsToMetadata = Maps.transformEntries(streams, (id, stream) -> {
+            MessageDigest digest = Sha256Hash.getMessageDigest();
+            InputStream compressedStream;
+            try {
+                // Hash the data before compression
+                compressedStream = new LZ4CompressingInputStream(new DigestInputStream(stream, digest));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            StreamMetadata metadata = storeBlocksAndGetFinalMetadata(t, id, compressedStream, false);
+            ByteString hashByteString = ByteString.copyFrom(digest.digest());
+            return StreamMetadata.newBuilder(metadata)
+                    .setHash(hashByteString)
+                    .build();
+        });
+        // Store the corrected metadata with the correct hashes
+        putMetadataAndHashIndexTask(t, idsToMetadata);
+
+        Map<Long, Sha256Hash> hashes = Maps.transformValues(idsToMetadata,
+                metadata -> new Sha256Hash(metadata.getHash().toByteArray()));
+        return hashes;
+    }
+
+    @Override
+    public InputStream loadStream(Transaction t, final Long id) {
+        return new LZ4BlockInputStream(super.loadStream(t, id));
+    }
+
+    @Override
+    public Map<Long, InputStream> loadStreams(Transaction t, Set<Long> ids) {
+        Map<Long, InputStream> compressedStreams = super.loadStreams(t, ids);
+        return Maps.transformValues(compressedStreams, stream -> {
+            return new LZ4BlockInputStream(stream);
+        });
+    }
+
+    @Override
+    protected void tryWriteStreamToFile(Transaction transaction, Long id, StreamMetadata metadata, FileOutputStream fos) throws IOException {
+        try (InputStream blockStream = makeStream(id, metadata);
```

Comment:
Does this compile? Looks like `AbstractGenericStreamStore.makeStream` is still private.